src/festim/exports/vtx.py

```diff
@@ -21,86 +40,141 @@ def __init__(self, filename: str | Path, ext: str) -> None:
 
     @property
     def filename(self):
         return self._filename
 
+    def is_it_time_to_export(self, current_time: float) -> bool:
+        """
+        Checks if the exported field should be written to a file or not based on the
+        current time and the times in `export.times`
+
+        Args:
+            current_time: the current simulation time
+
+        Returns:
+            bool: True if the exported field should be written to a file, else False
+        """
+
+        if self.times is None:
+            return True
+
+        for time in self.times:
+            if np.isclose(time, current_time, atol=0):
+                return True
+
+        return False
+
 
 class VTXTemperatureExport(ExportBaseClass):
-    def __init__(self, filename: str | Path):
-        super().__init__(filename, ".bp")
+    """Export temperature field functions to VTX file
+
+    Args:
+        filename: The name of the output file
+        times: if provided, the field will be exported at these timesteps. Otherwise
+            exports at all timesteps. Defaults to None.
+
+    Attributes:
+        filename: The name of the output file
+        times: if provided, the field will be exported at these timesteps. Otherwise
+            exports at all timesteps. Defaults to None.
+        writer: The VTXWriter object used to write the file
+    """
+
+    writer: io.VTXWriter
+
+    def __init__(
+        self,
+        filename: str | Path,
+        times: Optional[list[float] | list[int] | None] = None,
+    ):
+        super().__init__(filename, ".bp", times)
 
 
 class VTXSpeciesExport(ExportBaseClass):
-    """Export functions to VTX file
+    """Export species field functions to VTX file
 
     Args:
         filename: The name of the output file
         field: Set of species to export
-        subdomain: A field can be defined on multiple domains.
-            This arguments specifies what subdomains we export on.
-            If `None` we export on all domains.
-        checkpoint: If True, the export will be a checkpoint file
-            using adios4dolfinx and won't be readable by ParaView.
-            Default is False.
+        subdomain: A field can be defined on multiple domains. This arguments specifies
+            what subdomains we export on. If `None` we export on all domains.
+        checkpoint: If True, the export will be a checkpoint file using adios4dolfinx
+            and won't be readable by ParaView. Default is False.
+        times: if provided, the field will be exported at these timesteps. Otherwise
+            exports at all timesteps. Defaults to None.
+
+    Attributes:
+        filename: The name of the output file
+        field: Set of species to export
+        times: if provided, the field will be exported at these timesteps. Otherwise
+            exports at all timesteps. Defaults to None.
+        writer: The VTXWriter object used to write the file
     """
 
-    field: list[_Species]
```

Comment:
why remove this?

Reply:
Because its an argument, and the type hinting is already used in the init? But can change it back

Reply:
Just this is an argument, and the type hinting is already used in the init? But I can add it back if you want